cmd/tiller-proxy/internal/handler/handler.go

```diff
@@ -165,19 +168,19 @@ func (h *TillerProxy) CreateRelease(w http.ResponseWriter, req *http.Request, pa
 	log.Printf("Creating Helm Release")
 	chartDetails, ch, err := getChart(req, h.ChartClient)
 	if err != nil {
-		response.NewErrorResponse(errorCode(err), err.Error()).Write(w)
+		response.NewErrorResponse(errorCodeWithDefault(err, http.StatusUnprocessableEntity), err.Error()).Write(w)
```

Comment:
we want to return `http.StatusUnprocessableEntity` to specify that the error is related to the requested chart, not an "unknown" error.

Reply:
> errorCodeWithDefault

but that helper uses `errorCodeWithDefault` as fallback no? You are not enforcing it? 

Why doing `errorCodeWithDefault` instead of errorCode? How can we now handle 500 errors coming from Tiller?

Reply:
Note that Tiller doesn't reply with status code so we cannot translate them to HTTP errors. In this case what I am avoiding is to return a 500 to the user. A 500 means that something went wrong in the server side while an error here means that something is wrong with the chart that we are trying to install. It's not an "internal server error".

Reply:
It's probably not the case of the `getChart`, that function can fail if the chart is not retrievable or if the request is not properly formed (that shouldn't happen in any case) so we can use `errorCode` there.